assets/src/edit-story/components/panels/textStyle.js

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useState } from '@wordpress/element';
+import { __, _x } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { InputGroup, SelectMenu } from '../form';
+import { SimplePanel } from './panel';
+import getCommonValue from './utils/getCommonValue';
+
+function TextStylePanel({ selectedElements, onSetProperties }) {
+  const textAlign = getCommonValue(selectedElements, 'textAlign');
+  const letterSpacing = getCommonValue(selectedElements, 'letterSpacing');
+  const lineHeight = getCommonValue(selectedElements, 'lineHeight');
+  const padding = getCommonValue(selectedElements, 'padding') || '';
+  const [state, setState] = useState({
+    textAlign,
+    letterSpacing,
+    lineHeight,
+    padding,
+  });
+  useEffect(() => {
+    setState({ textAlign, letterSpacing, lineHeight, padding });
+  }, [textAlign, letterSpacing, lineHeight, padding]);
+  const handleSubmit = (evt) => {
+    onSetProperties(state);
+    evt.preventDefault();
+  };
+
+  const alignmentOptions = [
+    { name: __('Default', 'web-stories'), value: '' },
+    { name: __('Left', 'web-stories'), value: 'left' },
+    { name: __('Right', 'web-stories'), value: 'right' },
+    { name: __('Center', 'web-stories'), value: 'center' },
+    { name: __('Justify', 'web-stories'), value: 'justify' },
+  ];
+
+  return (
+    <SimplePanel
+      name="style"
+      title={__('Style', 'web-stories')}
+      onSubmit={handleSubmit}
+    >
+      <SelectMenu
+        label={__('Alignment', 'web-stories')}
+        options={alignmentOptions}
+        isMultiple={'' === textAlign}
```

Comment:
But if the empty string is actually one of the legal values of this dropdown, how do you know if it's just empty or actually the result of different values from a multiselection?

Reply:
@ndev1991 in order to move it to another PR, take a look at this improvement. 👍